config/routes.rb

```diff
@@ -2,5 +2,9 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+  resources :posts, only: [:new, :create]
+  resources :comments, only: [:new, :create]
+  resources :likes, only: [:new, :create]
+
```

Comment:
- [ ] **[OPTIONAL]** I think we can improve these routes to be nested properly. So that at every new or create action we can have access to the post and user IDs.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:new, :create]
  end

Reply:
thanks for the suggestion @TedLivist